docs/Makefile

```diff
@@ -48,13 +51,26 @@ help:
 	@echo "  doctest    to run all doctests embedded in the documentation (if enabled)"
 	@echo "  coverage   to run coverage check of the documentation (if enabled)"
 
+
+# function creates exit code for html target
+define report_result_html
+	if [ -s $(1) ]; \
+	then \
+		echo Build Failed. Check documentation_errors.log file for details.; \
+		exit 1; \
+	fi \
```

Comment:
This is just bash, not makefile syntax. Not that it's very important here as it's quite unlikely that we will have spaces in the `$(1)` var, but so you know, when doing variable expansions, you should be careful with the quotes:
```shell
if [[ -s "$(1)" ]]; then...
```
Specially with the `rm` commands (like the cleanup after this).

You can also remove the `\` of the `fi` line and remove the extra newline before the `endef`